storage/innobase/handler/handler0alter.cc

```diff
@@ -7228,8 +7231,7 @@ prepare_inplace_alter_table_dict(
 		ctx->trx->commit();
 		trx_start_for_ddl(ctx->trx, op);
 
-		if (!ctx->new_table->fts
-		    || ib_vector_size(ctx->new_table->fts->indexes) == 0) {
+		if (fts_instance_created) {
```

Comment:
Can you add some source code comments here to explain what this fix does and why it works? Where and why were the common tables being dropped before this fix, and why does that no longer happen ?